common/internal-types/src/tickets.rs

```diff
@@ -488,31 +488,38 @@ impl TryFrom<&[u8]> for Ticket {
 
     fn try_from(value: &[u8]) -> std::result::Result<Self, Self::Error> {
         if value.len() == Self::SIZE {
-            // TODO: not necessary to transmit over the wire, only the counterparty is sufficient
-            let channel_id = Hash::try_from(&value[0..32])?;
+            let mut offset = 0;
+
+            // TODO: not necessary to the ChannelId over the wire, only the counterparty is sufficient
+            let channel_id = Hash::try_from(&value[offset..offset + Hash::SIZE])?;
+            offset += Hash::SIZE;
+
             let mut amount = [0u8; 32];
-            amount[20..32].copy_from_slice(&value[Hash::SIZE..Hash::SIZE + 12]);
+            amount[20..32].copy_from_slice(&value[offset..offset + 12]);
+            offset += 12;
 
             let mut index = [0u8; 8];
-            index[2..8].copy_from_slice(&value[Hash::SIZE + 12..Hash::SIZE + 12 + 6]);
+            index[2..8].copy_from_slice(&value[offset..offset + 6]);
+            offset += 6;
 
             let mut index_offset = [0u8; 4];
-            index_offset.copy_from_slice(&value[Hash::SIZE + 12 + 6..Hash::SIZE + 12 + 6 + 4]);
+            index_offset.copy_from_slice(&value[offset..offset + 4]);
+            offset += 4;
 
             let mut channel_epoch = [0u8; 4];
-            channel_epoch[1..4].copy_from_slice(&value[Hash::SIZE + 12 + 6 + 4..Hash::SIZE + 12 + 6 + 4 + 3]);
+            channel_epoch[1..4].copy_from_slice(&value[offset..offset + 3]);
+            offset += 3;
 
             let mut encoded_win_prob = [0u8; 7];
-            encoded_win_prob.copy_from_slice(&value[Hash::SIZE + 12 + 6 + 4 + 3..Hash::SIZE + 12 + 6 + 4 + 3 + 7]);
+            encoded_win_prob.copy_from_slice(&value[offset..offset + 7]);
+            offset += 7;
```

Comment:
Wouldn't it be better to have all these constants named... and have the `ENCODED_TICKET_LENGTH` compile time assert against those `pub const` values? That way a breaking build could not happen, unless the compile check is updated as well. Or a unit test.

Reply:
This is already tested, left the debug assert there to better fail in that unit test.